### PR TITLE
[Java Time] Covid certs (EXPOSUREAPP-2846)

### PR DESCRIPTION
**Pull request by mtwalli**
## Refactoring Ticket
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-2846

### CODE REVIEW

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/common/DateFormattingTest.kt [hunk 1]
```diff
@@ -29,12 +29,12 @@ class DateFormattingTest : BaseTest() {
 
     @Test
     fun `format date time`() {
-        val tz = DateTimeZone.forID("Europe/Berlin")
+        val tz = ZoneId.of("Europe/Berlin")
         "2021-08-20T10:03:12Z".formatDateTime(tz) shouldBe "2021-08-20 12:03 UTC +02"
         "2021-08-20T12:03:12+02".formatDateTime(tz) shouldBe "2021-08-20 12:03 UTC +02"
-        "2021-08-20T12:03:12+0200".formatDateTime(tz) shouldBe "2021-08-20 12:03 UTC +02"
```
Comment (chiljamgossow) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/common/DateFormattingTest.kt:35]: As far as I remember, is was requested to have this specific output with UTC in it

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/person/core/PersonCertificatesStorageTest.kt [hunk 2]
```diff
@@ -517,11 +517,11 @@ class PersonCertificatesStorageTest : BaseTest() {
...
                         "{\"dateOfBirth\":\"20.10.2020\",\"familyNameStandardized\":\"llNN\",\"givenNameStandardized\":\"ffNN\"}": {
                           "lastSeenBoosterRuleIdentifier": "BRN-456",
-                          "lastBoosterNotifiedAt": 0,
+                          "lastBoosterNotifiedAt": 0.0,
                           "showDccReissuanceBadge": false,
-                          "lastDccReissuanceNotifiedAt": 0,
+                          "lastDccReissuanceNotifiedAt": 0.0,
                           "showAdmissionStateChangedBadge": false,
-                          "lastAdmissionStateNotifiedAt": 0
+                          "lastAdmissionStateNotifiedAt": 0.0
```
Comment (chiljamgossow) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/person/core/PersonCertificatesStorageTest.kt:524]: Has the data type changed here?

Comment (mtwalli) [reply Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/person/core/PersonCertificatesStorageTest.kt:524]: yes, it is not Java Instant not Joda Instant. Serialization is done by `JavaTimeModule` in jackson. not sure why is `0.0` not `0`.

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/common/DateFormattingTest.kt (hunk 1 above)
Comment (mtwalli) [reply Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/common/DateFormattingTest.kt:35]: yes, I had a look at the specs, the time zone in this example does not parse in java.

Comment (mtwalli) [reply Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/common/DateFormattingTest.kt:35]: see https://github.com/corona-warn-app/cwa-app-android/pull/5593/commits/8d5855a330ab7d43a86fd32f3abe04a2be6cedbd

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/censors/dcc/DccQrCodeCensor.kt [hunk 3]
```diff
@@ -77,7 +79,7 @@ class DccQrCodeCensor @Inject constructor() : BugCensor {
         dates += validUntilFormatted
 
         validUntil?.let {
-            dates += it.toShortDayFormat()
+            dates += it.format(DateTimeFormatter.ofLocalizedDate(FormatStyle.SHORT))
```
Comment (schauersbergern) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/censors/dcc/DccQrCodeCensor.kt:82]: Why not add this to JavaTimeAndDateExtenstions?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/censors/dcc/DccQrCodeCensor.kt [hunk 4]
```diff
@@ -101,7 +103,7 @@ class DccQrCodeCensor @Inject constructor() : BugCensor {
 
         dates += sampleCollectedAtFormatted
         sampleCollectedAt?.let {
-            dates += it.toShortDayFormat()
+            dates += it.toLocalDateUtc().format(DateTimeFormatter.ofLocalizedDate(FormatStyle.SHORT))
```
Comment (schauersbergern) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/censors/dcc/DccQrCodeCensor.kt:106]: Why not add this to JavaTimeAndDateExtenstions?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/pdf/core/DccExportFilter.kt [hunk 5]
```diff
@@ -30,8 +29,8 @@ internal fun List<CwaCovidCertificate>.sort(): List<CwaCovidCertificate> = sorte
         {
             when (it) {
                 is TestCertificate -> it.sampleCollectedAt
-                is VaccinationCertificate -> it.vaccinatedOn?.toInstantMidnightUtc()
-                is RecoveryCertificate -> it.testedPositiveOn?.toInstantMidnightUtc()
+                is VaccinationCertificate -> it.vaccinatedOn?.atStartOfDay(ZoneOffset.UTC)?.toInstant()
```
Comment (schauersbergern) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/pdf/core/DccExportFilter.kt:32]: Why not add this to JavaTimeAndDateExtenstions?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/items/CwaUserCard.kt [hunk 6]
```diff
@@ -56,7 +57,7 @@ internal fun formatBirthDate(dateOfBirthFormatted: String): String =
     try {
         // Formatted dob should be in the form of `yyyy-MM-dd` -> short day format like any other date in the screen
         // if failed display the original provided value , please note that dob can be YYYY-MM or YYYY only
-        dateOfBirthFormatted.parseLocalDate()?.toShortDayFormat()!!
+        dateOfBirthFormatted.parseLocalDate()?.format(DateTimeFormatter.ofLocalizedDate(FormatStyle.SHORT))!!
```
Comment (schauersbergern) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/items/CwaUserCard.kt:60]: Why not add this to JavaTimeAndDateExtenstions?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/core/model/CoronaTest.kt [hunk 7]
```diff
@@ -63,7 +63,7 @@ data class CoronaTest(
     }
 
     fun getFormattedRegistrationDate(): String =
-        registeredAt.toUserTimeZone().toLocalDate().format(DateTimeFormatter.ofLocalizedDate(FormatStyle.SHORT))
+        registeredAt.toLocalDateTimeUserTz().toLocalDate().format(DateTimeFormatter.ofLocalizedDate(FormatStyle.SHORT))
```
Comment (SamuraiKek) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/core/model/CoronaTest.kt:66]: Wouldn't it be better to have a `toUserTimeZone` extension function like we had before in the `TimeAndDateExtensions`? I see we already have a `JavaTimeAndDateExtensions` file.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/censors/dcc/DccQrCodeCensor.kt (hunk 3 above)
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/censors/dcc/DccQrCodeCensor.kt:82]: Well, I prefer to use the standard way directly , I will had to look into the implementation of these functions to see what format is uses.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/core/model/CoronaTest.kt (hunk 7 above)
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/core/model/CoronaTest.kt:66]: https://github.com/corona-warn-app/cwa-app-android/pull/5593#discussion_r982141284

**Review by SamuraiKek — APPROVED** [review]
Looks good 👍

**Review by schauersbergern — APPROVED** [review]

**Pull request merged**